.github/workflows/publish_dev_artifacts.yml

```diff
@@ -57,20 +56,15 @@ jobs:
         # We use "mr" instead of "pr" to denote pull request builds, as this prefix comes before "nightly" when lexically
         # sorting packages by version. This means that when installing the package without specifying a version the
         # nighly version is considered more current than mr versions and installed by default
-        run: sed -i -e 's/^version = "\([0-9]\+\.[0-9]\+\.[0-9]\+\).*/version = "\1-mr${{ github.event.number }}"/' server/Cargo.toml
+        run: sed -i -e 's/^version = "\([0-9]\+\.[0-9]\+\.[0-9]\+\).*/version = "\1-mr${{ github.event.number }}"/' rust/operator-binary/Cargo.toml
       - name: Build
         run: ~/.cargo/bin/cargo +nightly build --verbose --release
       - name: Build RPM package
-        run: server/packaging/buildrpm.sh stackable-kafka-operator-server
-      - name: Check workflow permissions
-        id: check_permissions
-        uses: scherermichael-oss/action-has-permission@1.0.6
```

Comment:
Remind me again why we don't need this step anymore?

Reply:
It doesn't work for pull requests from forks.
Not sure if the link works, but I've tried to link to a workflow run below that shows this check not correctly skipping the publish artifacts step.

https://github.com/stackabletech/zookeeper-operator/pull/184/checks?check_run_id=3533503667

I've replaced the action with the following in the publish step, which seems to work better:

```
        env:
          NEXUS_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
        if: env.NEXUS_PASSWORD != null
```